sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcTestDataSet.java

```diff
@@ -0,0 +1,101 @@
+   * mvn test-compile exec:java -Dexec.mainClass=org.apache.beam.sdk.io.jdbc.JdbcTestDataSet \
+   *   -Dexec.args="--postgresIp=1.1.1.1 --postgresUsername=postgres --postgresDatabaseName=myfancydb \
+   *   --postgresPassword=yourpassword --postgresSsl=false" \
+   *   -Dexec.classpathScope=test
+   * @param args Please pass options from PostgresTestOptions used for connection to postgres as shown above.
+   */
+  public static void main(String[] args) throws SQLException {
+    PGSimpleDataSource dataSource = new PGSimpleDataSource();
+    PipelineOptionsFactory.register(PostgresTestOptions.class);
+    PostgresTestOptions options = PipelineOptionsFactory.fromArgs(args).create().as(PostgresTestOptions.class);
+
+    dataSource.setDatabaseName(options.getPostgresDatabaseName());
+    dataSource.setServerName(options.getPostgresIp());
+    dataSource.setPortNumber(options.getPostgresPort());
+    dataSource.setUser(options.getPostgresUsername());
+    dataSource.setPassword(options.getPostgresPassword());
+    dataSource.setSsl(options.getPostgresSsl());
+
+    createReadDataTable(dataSource);
+  }
+
+  public final static String READ_TABLE_NAME = "BEAMTESTREAD";
+
+  public static void createReadDataTable(DataSource dataSource) throws SQLException {
+    createDataTable_(dataSource, READ_TABLE_NAME);
+  }
+
+  public static String createWriteDataTable(DataSource dataSource) throws SQLException {
+    String tableName = "BEAMTEST".concat(new Long(new Date().getTime()).toString());
+    createDataTable_(dataSource, tableName);
+    return tableName;
+  }
+
+  public static void createDataTable_(DataSource dataSource, String tableName) throws SQLException {
+    try (Connection connection = dataSource.getConnection()) {
+      // something like this will need to happen in tests on a newly created postgres server,
+      // but likely it will happen in perfkit, not here
+      // alternatively, we may have a pipelineoption indicating whether we want to
+      // re-use the database or create a new one
+      try (Statement statement = connection.createStatement()) {
+        statement.execute(
+            String.format("create table %s (id INT, name VARCHAR(500))", tableName));
+      }
+
+      String[] scientists = {"Einstein", "Darwin", "Copernicus", "Pasteur", "Curie", "Faraday",
+          "Newton", "Bohr", "Galilei", "Maxwell"};
+      connection.setAutoCommit(false);
+      try (PreparedStatement preparedStatement =
+               connection.prepareStatement(
+                   String.format("insert into %s values (?,?)", tableName))) {
+        for (int i = 0; i < 1000; i++) {
+          int index = i % scientists.length;
+          preparedStatement.clearParameters();
+          preparedStatement.setInt(1, i);
+          preparedStatement.setString(2, scientists[index]);
+          preparedStatement.executeUpdate();
+        }
+      }
+      connection.commit();
+    }
+
+    System.out.println("Created table " + tableName);
```

Comment:
Use a logger, and assume the environment will set up an appropriate SLF4J dependency